src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -2408,5 +2545,201 @@ static void M(I i)
                 //         var s1 = i.F1();
                 Diagnostic(ErrorCode.ERR_InteropStructContainsMethods, "i.F1()").WithArguments("S1").WithLocation(6, 18));
         }
+
+        [Fact]
+        public void ImplicitlyInitializedField_Simple()
+        {
+            var source = @"
+struct S
+{
+    public int x;
+    public S() { }
+}";
+            CreateCompilation(source, parseOptions: TestOptions.Regular10)
+                .VerifyDiagnostics(
+                // (4,16): warning CS0649: Field 'S.x' is never assigned to, and will always have its default value 0
+                //     public int x;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "x").WithArguments("S.x", "0").WithLocation(4, 16),
+                // (5,12): error CS0171: Field 'S.x' must be fully assigned before control is returned to the caller
+                //     public S() { }
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S").WithArguments("S.x").WithLocation(5, 12),
+                // (5,12): error CS8652: The feature 'implicit initialization in struct constructors' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public S() { }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "S").WithArguments("implicit initialization in struct constructors").WithLocation(5, 12));
+
+            CreateCompilation(source, options: TestOptions.DebugDll.WithReportSuppressedDiagnostics(true))
+                .VerifyDiagnostics(
+                    // (4,16): warning CS0649: Field 'S.x' is never assigned to, and will always have its default value 0
+                    //     public int x;
+                    Diagnostic(ErrorCode.WRN_UnassignedInternalField, "x").WithArguments("S.x", "0").WithLocation(4, 16),
+                    // (5,12): warning CS8881: Field 'S.x' must be fully assigned before control is returned to the caller
+                    //     public S() { }
+                    Diagnostic(ErrorCode.WRN_UnassignedThis, "S", isSuppressed: true).WithArguments("S.x").WithLocation(5, 12));
+
+            // TODO: how are suppressions removed from compiler diagnostics?
```

Comment:
@mavasani could you please let me know if I'm on the right track re: compiler diagnostics which are suppressed by default? My expectation is that some kind of project-level configuration should cause suppressed diagnostics to be displayed?

Reply:
> some kind of project-level configuration should cause suppressed diagnostics to be displayed

So suppressions is a very broad term that gets used around in different contexts, so I'd want to know which one you are talking about:
1. Disabled by default diagnostic: An analyzer or compiler diagnostic descriptor constructor takes a `bool isEnabledByDefault` parameter, which is specified as false means the diagnostic is never reported, regardless of the value specified for [CompilationWithAnalyzersOptions.ReportSuppressedDiagnostics](https://sourceroslyn.io/Microsoft.CodeAnalysis/R/07dfda34cf4b577c.html). Such diagnostics need an explicit editorconfig or ruleset entry that specifies a hidden/info/warning/error severity to get enabled and reported. This would need to be a warning/error severity to get reported on command line build output.
2. Reported diagnostic that is suppressed, either in source (via pragmas, SuppressMessageAttribute, etc.) or programmatically (via a DiagnosticSuppressor): Both these approaches would lead to a diagnostic with `IsSuppressed` property being true. There is no way to un-suppress such diagnostics. However, [CompilationWithAnalyzersOptions.ReportSuppressedDiagnostics](https://sourceroslyn.io/Microsoft.CodeAnalysis/R/07dfda34cf4b577c.html) can be set to true by analyzer hosts to allow these suppressed diagnostics to be passed onto the hosts, so they can show it to users if desired. IDE analyzer host does this explicitly so that `SuppressionState` column in Error list can be configured to show such suppressed diagnostics to users. There is no way to have command line compiler dump suppressed diagnostics (see [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis/CommandLine/CommonCompiler.cs,577)). Note that the `ErrorLogger` does log suppress diagnostics to error log file - see [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis/CommandLine/SarifV2ErrorLogger.cs,66).
3. Compiler diagnostics which are `Hidden` severity by default - Though these are technically not suppressed diagnostics, but from the perspective of command line builds, these are kind of suppressed by default. These are always reported in all analyzer hosts except the command line builds, for example [report unused usings](https://sourceroslyn.io/#Microsoft.CodeAnalysis/Compilation/Compilation.cs,2455) reports hidden diagnostics for IDE to consume and gray out these usings. Users can specify an editorconfig or ruleset entry with severity warning/error to see these diagnostics on command line builds.

Reply:
I looked at your PR a bit, and I feel you should take approach 1. (disabled by default diagnostic) or 3. (hidden by default diagnostic) as you do want users to be able to turn this diagnostic on and see on command line builds. 3 is more appropriate if you feel analyzer hosts, such as IDE live analysis, would want to do something with these hidden diagnostics in the default configuration, such as say show code fixes. Otherwise, 1 would be a better approach. Currently, all compiler diagnostics are always enabled by default, you would need to tweak [this code](https://sourceroslyn.io/#Microsoft.CodeAnalysis/Diagnostic/DiagnosticInfo.cs,93) to compute `isEnabledByDefault` based on error code to go this route.

Reply:
Thanks so much for the insight. I don't anticipate analyzers/fixes keying off these diagnostics. If the diagnostics were deliberately enabled, perhaps an analyzer might want to insert an explicit 'default' assignment to a field, for example, but it seems very niche. Therefore, I'll try moving forward with approach (1).

Reply:
I played with approach (1) today and found a few issues:

1. _The diagnostics never show in the editor_, even when .editorconfig includes e.g. `dotnet_diagnostic.<my-error-code>.severity = error`.
2. _The diagnostics show by default in the command line builds._ The severity does match the .editorconfig, and when the severity is explicitly set to 'none' then no diagnostic appears in the build.

Tomorrow I'll try to investigate these issues further. Perhaps they are already known or I can figure out how to solve separately/in advance of merging this PR.

Have pushed my work in 87422e8edfe8981c61849bf89697421fc61c673f. Loads of additional test baselines do need to be updated.

Reply:
Have resolved the issues I encountered. Mainly, we needed to remove the assumption that compiler diagnostics are always `isEnabledByDefault: true`.